src/main/java/bio/terra/service/filedata/azure/blobstore/AzureBlobStorePdao.java

```diff
@@ -72,24 +72,15 @@ public FSFileInfo copyFile(
         profileDao.getBillingProfileById(fileLoadModel.getProfileId());
 
     BlobContainerClientFactory destinationClientFactory =
-        getTargetDataClientFactory(profileModel, storageAccountResource, false);
+        getTargetDataClientFactory(profileModel, storageAccountResource, ContainerType.DATA, false);
 
-    BlobUrlParts blobUrl = BlobUrlParts.parse(fileLoadModel.getSourcePath());
-    BlobContainerClientFactory sourceClientFactory;
-    if (isSignedUrl(fileLoadModel.getSourcePath())) {
-      sourceClientFactory = getSourceClientFactory(fileLoadModel.getSourcePath());
-    } else {
-      // Use application level authentication
-      sourceClientFactory =
-          getSourceClientFactory(
-              blobUrl.getAccountName(),
-              resourceConfiguration.getAppToken(profileModel.getTenantId()),
-              blobUrl.getBlobContainerName());
-    }
+    BlobContainerClientFactory sourceClientFactory =
+        buildSourceClientFactory(profileModel.getTenantId(), fileLoadModel.getSourcePath());
 
     BlobCrl blobCrl = getBlobCrl(destinationClientFactory);
```

Comment:
The process of wiring up an instance of `BlobCrl` from the billing profile may be useful elsewhere. Does it make sense to create constructor of the `BlobCrl` that accepts a `BillingProfileModel`?